### ISSUE TITLE
Support full URL format as overload for creating sessions.

### ISSUE THREAD

**Issue by Ijwu** [ArchipelagoMW/Archipelago.MultiClient.Net#62]
Should support taking in one URL string to attempt to parse.

Format is: [scheme][<user>[:<password>]@]<address>:<port>

If `scheme` is not provided or is not `ws://` then it should be replaced with `ws://`. If `user` (and subsequently, optionally, password) is provided then perhaps we can cache that and allow for a streamlined connection process?

**Comment by Berserker66** [comment]
in race cases scheme could also be wss://

**Comment by Jarno458** [comment]
yeah we should support wss:// for people who do host it on a secure connection

**Comment by Jarno458** [comment]
i just tryed this myzelf it seems the webhost link is archipelago://Jarno2:@archipelago.gg:38281
so the scheme has to be replaced than still

**Comment by Jarno458** [comment]
While i can see this as a interesting thing to add it currently does not match our pattern
ArchipelagoSessionFactory.CreateSession(Uri uri) could totally support it but,
session.LoginAsync(string game, string name, ItemsHandlingFlags itemsHandlingFlags, Version version = null, string[] tags = null, string uuid = null, string password = null)
will be a problem. what do we supply as name than? make it optional?

**Comment by Ijwu** [comment]
> While i can see this as a interesting thing to add it currently does not match our pattern ArchipelagoSessionFactory.CreateSession(Uri uri) could totally support it but, session.LoginAsync(string game, string name, ItemsHandlingFlags itemsHandlingFlags, Version version = null, string[] tags = null, string uuid = null, string password = null) will be a problem. what do we supply as name than? make it optional?

![image](https://user-images.githubusercontent.com/1393017/200343340-06e31122-7c75-4e33-911a-12dd4d4340f4.png)

**Comment by Jarno458** [comment]
Yes, i was more wondering how we should structure the code, 
Should `ArchipelagoSessionFactory` have a new method that both sets up a session and connects?
and then possiably also the async over loads and a way to handle the logic results...

I dont think this should be a responseability of the SessionFactory, 
i could have `ArchipelagoSessionFactory.CreateSession(Uri uri)`
and `session.LoginAsync(string game, Uri uri, ItemsHandlingFlags itemsHandlingFlags, Version version = null, string[] tags = null, string uuid = null)`
so to have an overload that supports the Uri, and removes the password
